src/Confluent.Kafka/Producer.cs

```diff
@@ -293,124 +309,323 @@ public Producer(IEnumerable<KeyValuePair<string, object>> config, bool manualPol
             }
         }
 
-        // TODO: think about Poll and flushing.
+        /// <summary>
+        ///     Initializes a new Producer instance.
+        /// </summary>
+        /// <param name="config">
+        ///     librdkafka configuration parameters (refer to https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
+        ///     Topic configuration parameters are specified via the "default.topic.config" sub-dictionary config parameter.
+        /// </param>
+        public Producer(IEnumerable<KeyValuePair<string, object>> config)
+            : this(config, false, false) {}
+
 
-        public void Poll(int millisecondsTimeout)
+        /// <summary>
+        ///     Poll for callback events. You will not typically need
```

Comment:
my terminology might not be quite right but will leave for now. delegate isn't right here i don't think.